src/main/java/org/ibp/api/brapi/v2/germplasm/BreedingMethodResourceBrapi.java

```diff
@@ -0,0 +1,108 @@
+	@RequestMapping(value = "/{crop}/brapi/v2/breedingmethods", method = RequestMethod.GET)
+	@ResponseBody
+	public ResponseEntity<EntityListResponse<BreedingMethod>> getAllBreedingMethods(@PathVariable final String crop, @ApiParam(value = BrapiPagedResult.CURRENT_PAGE_DESCRIPTION)
+	@RequestParam(value = "page", required = false) final Integer currentPage,
+		@ApiParam(value = BrapiPagedResult.PAGE_SIZE_DESCRIPTION)
+		@RequestParam(value = "pageSize", required = false) final Integer pageSize) {
+
+		final int finalPageNumber = currentPage == null ? BrapiPagedResult.DEFAULT_PAGE_NUMBER : currentPage;
+		final int finalPageSize = pageSize == null ? BrapiPagedResult.DEFAULT_PAGE_SIZE : pageSize;
+
+		final BreedingMethodSearchRequest searchRequest = new BreedingMethodSearchRequest();
+		final PagedResult<BreedingMethodDTO> resultPage = new PaginatedSearch().executeBrapiSearch(finalPageNumber, finalPageSize,
+			new SearchSpec<BreedingMethodDTO>() {
+
+				@Override
+				public long getCount() {
+					return BreedingMethodResourceBrapi.this.breedingMethodService.countBreedingMethods(searchRequest);
+				}
+
+				@Override
+				public List<BreedingMethodDTO> getResults(final PagedResult<BreedingMethodDTO> pagedResult) {
+					return BreedingMethodResourceBrapi.this.breedingMethodService.getBreedingMethods(crop, searchRequest, new PageRequest(finalPageNumber, finalPageSize));
+				}
+			});
+
+		final ModelMapper modelMapper = BreedingMethodMapper.getInstance();
+		final List<BreedingMethod> breedingMethods = new ArrayList<>();
+		for (final BreedingMethodDTO dto : resultPage.getPageResults()) {
+			breedingMethods.add(modelMapper.map(dto, BreedingMethod.class));
+		}
+		final Result<BreedingMethod> result = new Result<BreedingMethod>().withData(breedingMethods);
+		final Pagination pagination = new Pagination().withPageNumber(resultPage.getPageNumber()).withPageSize(resultPage.getPageSize())
+			.withTotalCount(resultPage.getTotalResults()).withTotalPages(resultPage.getTotalPages());
+		final Metadata metadata = new Metadata().withPagination(pagination);
+		final EntityListResponse<BreedingMethod> entityListResponse = new EntityListResponse<BreedingMethod>(metadata, result);
+		return new ResponseEntity<>(entityListResponse, HttpStatus.OK);
+	}
+
+	@ApiOperation(value = "Get the details of a specific Breeding Method used to produce Germplasm", notes = "Get the details of a specific Breeding Method")
+	@RequestMapping(value = "/{crop}/brapi/v2/breedingmethod/{breedingMethodDbId}", method = RequestMethod.GET)
+	@ResponseBody
+	public ResponseEntity<SingleEntityResponse<BreedingMethod>> getBreedingMethodById(
+		@PathVariable final String crop,
+		@PathVariable final Integer breedingMethodDbId) {
+
+		final BreedingMethodDTO breedingMethodDTOS = this.breedingMethodService.getBreedingMethod(breedingMethodDbId);
+		if (Objects.isNull(breedingMethodDTOS)) {
```

Comment:
Move validation of invalid method id to ` this.breedingMethodService.getBreedingMethod`